url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+      {
+         uaSquareIndex = Long.numberOfTrailingZeros(uaBbOccupiedTemp);
+         uaBbAttackerSquare = bbSquare[uaSquareIndex];
+         
+         uaBbAttacked = bbKnightAttack[uaSquareIndex];
+         bbAttackedFromSquare[uaSquareIndex] |= uaBbAttacked;
+         
+         updateAttackingTo(uaBbAttackerSquare, uaBbAttacked);
+         
+         uaBbOccupiedTemp ^= uaBbAttackerSquare;
+      }
+      
+      //Se generan las casillas atacadas por las torres blancas
+      uaBbOccupiedTemp = bbPiecesOccupation[Piece.WHITE_ROOK.index];
+      uaNumPieces = Long.bitCount(uaBbOccupiedTemp);
+      for(int i = 0; i < uaNumPieces; i++)
+      {
+         uaSquareIndex = Long.numberOfTrailingZeros(uaBbOccupiedTemp);
+         uaBbAttackerSquare = bbSquare[uaSquareIndex];
+         
+         //Se generan las casillas atacadas en filas y columnas
+         uaBbAttacked = getAttackedInRank(uaSquareIndex);
+         uaBbAttacked |= getAttackedInFile(uaSquareIndex);
+         bbAttackedFromSquare[uaSquareIndex] |= uaBbAttacked;
+         
+         updateAttackingTo(uaBbAttackerSquare, uaBbAttacked);
+         
+         uaBbOccupiedTemp ^= uaBbAttackerSquare;
+      }
+      
+      //Se generan las casillas atacadas por las torres negras
+      uaBbOccupiedTemp = bbPiecesOccupation[Piece.BLACK_ROOK.index];
+      uaNumPieces = Long.bitCount(uaBbOccupiedTemp);
+      for(int i = 0; i < uaNumPieces; i++)
+      {
+         uaSquareIndex = Long.numberOfTrailingZeros(uaBbOccupiedTemp);
+         uaBbAttackerSquare = bbSquare[uaSquareIndex];
+         
+         //Se generan las casillas atacadas en filas y columnas
+         uaBbAttacked = getAttackedInRank(uaSquareIndex);
+         uaBbAttacked |= getAttackedInFile(uaSquareIndex);
+         
+         bbAttackedFromSquare[uaSquareIndex] |= uaBbAttacked;
+         
+         updateAttackingTo(uaBbAttackerSquare, uaBbAttacked);
+         
+         uaBbOccupiedTemp ^= uaBbAttackerSquare;
+      }
+      
+      //Se generan las casillas atacadas por las reinas blancas
+      uaBbOccupiedTemp = bbPiecesOccupation[Piece.WHITE_QUEEN.index];
+      uaNumPieces = Long.bitCount(uaBbOccupiedTemp);
+      for(int i = 0; i < uaNumPieces; i++)
+      {
+         uaSquareIndex = Long.numberOfTrailingZeros(uaBbOccupiedTemp);
+         uaBbAttackerSquare = bbSquare[uaSquareIndex];
+         
+         //Se generan las casillas atacadas en filas y columnas.
+         uaBbAttacked = getAttackedInRank(uaSquareIndex);
+         uaBbAttacked |= getAttackedInFile(uaSquareIndex);
+         //Se generan las casillas atacadas en diagonal
+         uaBbAttacked |= getAttackedInMainDiagonal(uaSquareIndex);
+         uaBbAttacked |= getAttackedInSecondaryDiagonal(uaSquareIndex);
+         
+         bbAttackedFromSquare[uaSquareIndex] |= uaBbAttacked;
+         
+         updateAttackingTo(uaBbAttackerSquare,uaBbAttacked);
+         
+         uaBbOccupiedTemp ^= uaBbAttackerSquare;
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1187:1204, 1209:1226.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.